sortinghat/parsing/grimoirelab.py

```diff
@@ -0,0 +1,358 @@
+        'gerrit', 'github', 'jira', 'mediawiki', 'meetup', 'phabricator',
+        'redmine', 'stackexchange', 'irc', 'telegram']
+
+
+class GrimoireLabParser(object):
+    """Parse identities and organizations using GrimoireLab format.
+
+    The GrimoireLab data format is a YAML stream that contains information
+    about unique identities and organizations.
+
+    The unique identities are stored in an object named 'uidentities'.
+    The keys of this object are the UUID of the unique identities.
+    Each unique identity object stores a list of identities and
+    enrollments.
+
+    Organizations are stored in 'organizations' object. Its keys
+    are the name of the organizations and each organization object is
+    related to a list of domains.
+
+    :param stream: stream to parse
+
+    :raises InvalidFormatError: raised when the format of the stream is
+        not valid.
+    """
+
+    EMAIL_ADDRESS_REGEX = r"^(?P<email>[^\s@]+@[^\s@.]+\.[^\s@]+)$"
+
+
+    def __init__(self, identites, domain_employer=None,
+                                        source='grimoirelab'):
+        self._identities = {}
+        self._organizations = {}
+        self.source = source
+
+        self.__parse(identites, domain_employer)
+
+    @property
+    def identities(self):
+        uids = [u for u in self._identities.values()]
+        uids.sort(key=lambda u: u.uuid)
+        return uids
+
+    @property
+    def organizations(self):
+        orgs = [o for o in self._organizations.values()]
+        orgs.sort(key=lambda o: o.name)
+        return orgs
+
+    def __parse(self, identities_stream, domain_employer_stream):
+        """Parse Eclipse stream"""
+
+        if not identities_stream:
+            raise InvalidFormatError(cause="stream cannot be empty or None")
+
+        self.__parse_organizations(domain_employer_stream)
+        self.__parse_identities(identities_stream)
+
+    def __parse_identities(self, stream):
+        """Parse identities using GrimoireLab format.
+
+        The GrimoireLab identities format is a YAML document following a
+        schema similar to the example below. More information available
+        at https://github.com/bitergia/identities
+
+        - profile:
+            name: Vivek K.
+            is_bot: false
+          github:
+            - vsekhark
+          email:
+            - vivek@****.com
+          enrollments:
+            - organization: Community
+              start: 1900-01-01
+              end: 2100-01-01
+
+        :parse json: YAML object to parse
+
+        :raise InvalidFormatError: raised when the format of the YAML is
+            not valid.
+        """
+        yaml = self.__load_yml(stream)
+
+        try:
+            for yid in yaml:
+
+                name = None
+                first_email = None
+                first_username = None
+                uuid = ''
+                is_bot = False
+
+                if "name" in yid["profile"]:
```

Comment:
I think you can compress this code with a call to `get()` method. For instance:
```
name = yid.get('profile', None)
```
It can be applied to the rest of variables like this one.